### PR TITLE
Added test for SSPI authentication

### PR DESCRIPTION
**Pull request by vlastahajek**
Added E2E test verifying fix in PR #483 (issue #450)

### CODE REVIEW

**Comment by kardianos** [comment]
Thanks. 
 * Do not use "github.com/stretchr/testify/require".
 * Put in an existing test file.
 * Use an existing helper to setup the connection string. See https://github.com/denisenkom/go-mssqldb/blob/2fea367d496dade570d9c1959be58a98a9061522/tds_test.go#L321 for how to modify the existing connection string.

**Comment by vlastahajek** [comment]
Thanks for quick review, @kardianos .
> * Do not use "github.com/stretchr/testify/require".

Can I ask why not? It makes test clear and  test doesn't look like demo code then.

> * Put in an existing test file.

Well, I wanted to to put it in `tds_test.go` at first, but it holds multi-platform tests, if i understand correctly. 
But SSPI will work only on Windows and tests what is inside sspi_windows.go, although it is not unit test. 

> * Use an existing helper to setup the connection string. See https://github.com/denisenkom/go-mssqldb/blob/2fea367d496dade570d9c1959be58a98a9061522/tds_test.go#L321

>    for how to modify the existing connection string.

I wanted to use it, but `makeConnStr` creates URL type connection string, whereas I need ADO style.

**Comment by vlastahajek** [comment]
I've just noticed that the require package makes build failed, on looking into it. Maybe, that was you point to remove this library ;)

**Comment by kardianos** [comment]
> Thanks for quick review, @kardianos .
> 
> > * Do not use "github.com/stretchr/testify/require".
> 
> Can I ask why not? It makes test clear and test doesn't look like demo code then.

 a. No other testing code uses testify or other testing helpers.
 b. Testing code is just code.
 c. If you want to reduce redundancy in testing, use table driven tests. Here is what I mean by that: https://dave.cheney.net/2019/05/07/prefer-table-driven-tests .

> > * Put in an existing test file.
> 
> Well, I wanted to to put it in `tds_test.go` at first, but it holds multi-platform tests, if i understand correctly.
> But SSPI will work only on Windows and tests what is inside sspi_windows.go, although it is not unit test.

If SSPI will only work on windows, then put in a standard testing file (say tds_test.go), but the first line should say:

```
if runtime.GOOS != "windows" {
    t.Skip("only on windows")
}
```

> 
> > * Use an existing helper to setup the connection string. See https://github.com/denisenkom/go-mssqldb/blob/2fea367d496dade570d9c1959be58a98a9061522/tds_test.go#L321
> 
> > for how to modify the existing connection string.
> 
> I wanted to use it, but `makeConnStr` creates URL type connection string, whereas I need ADO style.

You don't need an ADO style connection string. All styles of connection strings get decomposed into the same key-value pairs.

**Comment by vlastahajek** [comment]
I've made required changes. 

I'm also fan of data driven tests. However, with respect to platform dependency of this test, this would make the data driven test more complex.
If go has other mechanisms to distinguish platform dependent stuff (file naming convention, build instructions), runtime check is not, IMHO, so nice :)  

Require is just sub-package of testify. However, without a package dependency mechanism is not easy to add it.

It made sense to me to add also E2E test with different connection string, but as there is a test for different connection strings, it would be redundant.

File: tds_test.go [hunk 1]
```diff
@@ -520,3 +521,37 @@ func TestBadConnect(t *testing.T) {
...
+	if runtime.GOOS != "windows" {
+		t.Skip("Only on windows")
+	}
+	checkConnStr(t)
+	connStr := makeConnStr(t)
+	params := connStr.Query()
+	params.Set("Integrated Security", "sspi")
+	connStr.RawQuery = params.Encode()
+
+	conn, err := sql.Open("mssql", connStr.String())
```
Comment (kardianos) [inline tds_test.go]: Do not call this `conn`. This is not a database connection, this represents a database pool that hasn't made a single connection.

Prefer `db` or `pool`.

The reason why this name is important is people read examples and testing code and sometimes mis-understand what is going on, then go on to write incorrect or poor quality code (like creating a connection pool for every query).

File: tds_test.go [hunk 2]
```diff
@@ -520,3 +521,37 @@ func TestBadConnect(t *testing.T) {
...
+	if err != nil {
+		t.Error("Open failed", err)
+	}
+	defer conn.Close()
+
+	stmt, err := conn.Prepare("select 1, 'abc'")
```
Comment (kardianos) [inline tds_test.go]: Prepare offers very little benefit in SQL Server. Don't use it. 
Please use `db.QueryRow` directly.

**Comment by vlastahajek** [comment]
No problem, make sense. Fixed.

However, I copied that from the  `examples/simple/simple.go`. This should be also amended to the best-practices form.

**Pull request merged**